packages/cli/src/tools/generator/promptSync.ts

```diff
@@ -12,33 +12,45 @@
 
 import fs from 'fs';
 
+type Options = {
+  echo?: string;
+  ask?: string;
+  value?: string;
+  autocomplete?: string[] | Function;
+};
+
 const term = 13; // carriage return
 
 function create() {
   return prompt;
 
-  function prompt(ask, value, opts) {
+  function prompt(
+    ask?: string | Options,
+    value?: string | Options,
+    opts?: Options,
+  ) {
     let insert = 0;
     opts = opts || {};
 
-    if (Object(ask) === ask) {
+    if (typeof ask === 'object') {
       opts = ask;
       ask = opts.ask;
-    } else if (Object(value) === value) {
+    } else if (typeof value === 'object') {
       opts = value;
       value = opts.value;
     }
     ask = ask || '';
     const echo = opts.echo;
     const masked = 'echo' in opts;
 
-    const fd =
-      process.platform === 'win32'
-        ? process.stdin.fd
-        : fs.openSync('/dev/tty', 'rs');
+    let fd = fs.openSync('/dev/tty', 'rs');
+    if (process.platform === 'win32') {
+      // @ts-ignore MEMO: fd exists when win32
+      fd = process.stdin.fd;
```

Comment:
Can we bring back previous behavior? No need to reassign, a conditional assignment per platform is ok

Reply:
![image](https://user-images.githubusercontent.com/11006919/64614722-177ccc80-d414-11e9-8cd0-f62bfcbf7e22.png)

When I use Typescript, it saids ts error occurred.
And When I `@ts-ignore` it, It also tells error since we can not use `@ts-ignore` in ternary operator. 😭 
![image](https://user-images.githubusercontent.com/11006919/64615291-3a5bb080-d415-11e9-8d95-ebe51f69f6fb.png)

Reply:
What about
```ts
let fd;
if (win32) { something; } else { somethingElse; }
```

Reply:
The thing is I'd like to avoid touching the code in tools/generator too much, because it works and will be removed in a year or so, because we'll get rid of legacy init.

Reply:
@thymikee Thanks for comment!
I brought back previous behavior and make it work!

changes are in ↓ 👍 
8f83905